### ISSUE TITLE
Azure kubernetes version is now too old [bug]

### ISSUE THREAD

**Issue by danlester** [Quansight/qhub#825]
## Describe the bug
The default kubernetes version of `qhub init` is now deprecated by Azure (at least in 'Central US')

Get this error on `qhub deploy`:
```
Error: creating Managed Kubernetes Cluster "qhubitaz-dev" (Resource Group "qhubitaz-dev"): containerservice.ManagedClustersClient#CreateOrUpdate: Failure sending request: StatusCode=400 -- Original Error: Code="AgentPoolK8sVersionNotSupported" Message="Version 1.18.19 is not supported in this region. Please use [az aks get-versions] command to get the supported version list in this region. For more information, please check https://aka.ms/supported-version-list"
```

## How can we help?
We need to upgrade this default version number. 
Oldest supported now is 1.19.11, but probably worth trying a newer one to avoid it being deprecated for a while...

## Your environment
Azure cloud

**Comment by costrouc** [comment]
Yeah I think that would be helpful. I was hoping that we could solve this much like on digital ocean. https://github.com/Quansight/qhub/blob/main/qhub/provider/cloud/digital_ocean.py#L42. Idea would be to query some API for the available azure kubernetes versions. If possible we could maybe get the available instances as well?

**Comment by costrouc** [comment]
1. Add azure sdk as an optional dependency for qhub it is available on conda-forge https://anaconda.org/conda-forge/azure
2. Call the appropriate methods to populate the cloud provider methods.

**Comment by costrouc** [comment]
Important locations in the code:
 - https://github.com/Quansight/qhub/blob/main/qhub/initialize.py is where the qhub-config.py is generated
 - you can assume that these environment variables are available https://docs.qhub.dev/en/latest/source/installation/setup.html#microsoft-azure
 - we need to write a provider similar to https://github.com/Quansight/qhub/blob/main/qhub/provider/cloud/digital_ocean.py but `azure.py` which does not exist yet

You can test this out with qhub by running `qhub init azure`

**Comment by costrouc** [comment]
Assigning @nayaksuraj

**Comment by danlester** [comment]
Since first time Azure deployments are currently broken (even in the latest full release 0.3.12) I think it would make sense to create this fix off the last [release 0.3.12](https://github.com/Quansight/qhub/releases/tag/v0.3.12) instead of main. Then it will be easier to release as v0.3.13 and also merge into main for the next major release.

**Comment by nayaksuraj** [comment]
Working on this to get right Kubernetes version supported by Azure.

**Comment by nayaksuraj** [comment]
Raised a PR for this issue. Waiting for review from @costrouc

**Comment by trallard** [comment]
This issue was fixed with #846 can this be closed, please?
